crates/exex/src/exex.rs

```diff
@@ -70,116 +68,53 @@ impl<Node: FullNodeComponents> KakarotRollup<Node> {
         while let Some(notification) = self.ctx.notifications.next().await {
             // Check if the notification contains a committed chain.
             if let Some(committed_chain) = notification.committed_chain() {
-                // Commit the new chain state.
-                self.commit_block(&committed_chain).await?;
                 // Send a notification that the chain processing is finished.
                 //
                 // Finished height is the tip of the committed chain.
                 //
                 // The ExEx will not require all earlier blocks which can be pruned.
                 self.ctx.events.send(ExExEvent::FinishedHeight(committed_chain.tip().number))?;
 
-                // Build the Kakarot hint processor with the print transaction hint.
+                // Build the Kakarot hint processor.
                 let mut hint_processor = KakarotHintProcessor::default().build();
 
-                // Run the cairo programs corresponding to the paths
-                for path in &paths {
-                    // Load the cairo program from the file
-                    let program = std::fs::read(path)?;
-
-                    // Execute the Cairo program with the specified configuration and hint
-                    // processor.
-                    let res = cairo_run(&program, &config, &mut hint_processor)?;
-
-                    // Extract the execution trace
-                    let trace = res.relocated_trace.clone().unwrap_or_default();
-
-                    // Extract the relocated memory
-                    let memory = res
-                        .relocated_memory
-                        .clone()
-                        .into_iter()
-                        .map(|x| x.unwrap_or_default())
-                        .collect();
-
-                    // Extract the public and private inputs
-                    //
-                    // We want to store the public input in the database in order to use them to run
-                    // the prover
-                    let public_input = res.get_air_public_input()?;
-                    let private_input = res.get_air_private_input();
-
-                    // Commit the execution trace to the database
-                    self.commit_cairo_execution_traces(
-                        committed_chain.tip().number,
-                        trace,
-                        memory,
-                        public_input,
-                        private_input,
-                    )?;
-                }
-            }
-        }
+                // Load the cairo program from the file
+                let program = std::fs::read(PathBuf::from("../../cairo/programs/os.json"))?;
```

Comment:
If we put this program it means that from now on we only do tests on our OS and nothing else correct, like no small random programs? I had the impression that the approach with the Paths vector was perhaps more modular for the moment in the sense that we could enter whatever we wanted in the `start` to test lots of different programs (even if it meant only entering a vector of an element with the OS path if we only want to test that). But hardcoding a path in the function seems a bit restrictive to me, no?

Reply:
the idea is that from the exex pov you don't really want to test cairo programs in the context of the exex but hints. these hints should be tested but probable in standalone cairo_run, no need to have the whole exex setup

on the other hand, on the exex we want to make sure that the real program (yet under construction) is working well